### DISCUSSION TITLE
Comment cartographier les processus

### DISCUSSION THREAD

**Discussion by llagar910e** [dbarzin/mercator#1000] [Q&A]
Bonjour,
J'ai du mal à savoir comment bien décomposer les processus.

Voici un exemple de ce que je pense être correct, qu'en pensez-vous ?

Pour information, le backup, l'envoi et l'archivage peuvent parfois être dans le même script. Il y a aussi l'épuration des backups que je n'ai pas spécifiée ici et qui se fait dans le même script que l'envoi.
Plusieurs sauvegardes différentes peuvent être envoyées vers le NAS mais avec une durée de rétention différente, ça serait donc à mettre en description de l'opération et non de la tâche d'archivage puisqu'une tâche peut être utilisée par plusieurs opérations ?

Merci pour votre aide

![image](https://github.com/user-attachments/assets/92d5fec8-962c-498d-8cb9-b0fe4dc87ea8)

**Comment by lynch4fr** [comment]
Bonjour,
J'ai posé une question similaire sur les processus:  https://github.com/dbarzin/mercator/discussions/970 et visiblement l'outil va evoluer dans ce sens. Je vous laisse lire la reponse de @dbarzin 
Cordialement

**Reply by llagar910e** [reply]
Bonjour,

Merci pour votre retour. Je ne connais pas la modélisation BPM mais je pense que votre question est plus poussée que la mienne.
Ma question se résume assez simplement par :
Est-ce que je considère comme un processus en est bien un ? 
Est-ce que je considère comme une opération en est bien une ? 
Est-ce que je considère comme une tâche en est bien une ? 
Si non, en quoi est-ce que mon découpage est mauvais ?

Ce sont des concepts que je trouve subjectifs et qui sont difficiles à apréhender pour quelqu'un qui n'a pas l'habitude de décomposer ainsi

**Comment by dbarzin** [comment]
Vous soulevez des points importants sur la modélisation des processus. Pour répondre :

- Processus : Représente une séquence complète orientée vers un objectif métier.
- Opérations : Correspondent à des étapes spécifiques dans un processus.
- Tâches : Sont des actions unitaires et réutilisables.

Votre découpage semble cohérent, mais pour les cas où une tâche est utilisée dans plusieurs opérations avec des variations (ex. rétention des backups), il est en effet pertinent de spécifier les différences dans la description des opérations.

N'hésitez pas à explorer la discussion mentionnée (#970) et à partager des exemples concrets si vous souhaitez un retour plus précis.

**Reply by llagar910e** [reply]
Je vous remercie, je vais partir sur cette modélisation et je verrai si je rencontre des problèmes en cours de route

**Comment by j-ova** [comment]
Bonjour,

Je reprends cette discussion intéressante car nous cherchons aussi à modéliser nos processus métier, et cherchons en plus à les lier à nos applications et services. 

Nous ne sommes pas encore dans la démarche de cartographie type BPM : pour le moment nous essayons de trouver le bon équilibre entre une vue pertinente, et une granularité que nous pouvons maintenir. 

A ce titre, le fait que le seul lien possible entre le domaine "Système d'information" et le domaine "Applications" soit entre un processus et une application nous apparait comme limitant. Par exemple : 
- Notre ERP est une seule application composée de nombreux services
- De nombreux processus se reposent sur différents services de l'ERP
- En l'état, notre cartographie ressemblerait à "tous les processus liés à un seul objet"

Il nous semble pertinent de pouvoir rentrer plus en détail dans les liens entre le système d'information et les applications : 
- A minima, pouvoir gérer des liens entre les objets processus/opérations et applications/services
- Voire, si le modèle le permet, rendre possible la création de liens entre tous les objets du domaine "Système d'information" (processus/activités/opérations/tâches) et tous ceux du domaine "Applications" (applications/services/modules)

Est-ce une évolution qui vous semble pertinente ?

Merci :)

**Reply by dbarzin** [reply]
Il est déjà possible de lier un processus ou une opération à une application dans Mercator. Cependant, je comprends votre besoin d'une granularité plus fine dans la cartographie entre le domaine "Système d'information" et le domaine "Applications".

Mercator est principalement conçu pour la gestion des opérations IT, avec un focus sur le traitement des incidents, l'impact sur les processus, la criticité et le nombre d'utilisateurs affectés. Il n'est pas spécifiquement orienté vers la modélisation détaillée des processus et des opérations.

Pour répondre à vos besoins de modélisation plus détaillée, je vous recommande d'explorer des outils spécialisés dans ce domaine, tels que TOGAF ou UML. Ces frameworks offrent des capacités avancées pour cartographier les relations entre les processus métier et les applications, y compris les services et modules spécifiques.

**Reply by j-ova** [reply]
> Mercator est principalement conçu pour la gestion des opérations IT [...] Il n'est pas spécifiquement orienté vers la modélisation détaillée des processus et des opérations.

Je comprends que Mercator est avant tout conçu pour la gestion des opérations IT et qu’il n’a pas vocation à être un outil de modélisation avancée des processus métier.

Cela dit, notre besoin ne porte pas tant sur une modélisation exhaustive des processus que sur une capacité à mieux refléter la réalité opérationnelle de notre SI dans Mercator.

Même en regardant la problématique sous le prisme des opérations IT :
- Nos processus ne sont pas directement liés à des applications, mais à des services spécifiques au sein de ces applications (particulièrement notre ERP)
- Une granularité plus fine améliorerait l’analyse des incidents et de leurs effets sur les services utilisés par les processus.

Cette évolution permettrait d'enrichir le rôle actuel de Mercator en offrant une meilleure traçabilité des interactions SI-Processus, sans pour autant le repositionner comme un outil BPM.
Cela renforcerait l’intérêt de l’outil pour des équipes qui, comme la nôtre, cherchent à maintenir une cartographie exploitable avec vision d'ensemble de son SI.

Serait-il envisageable d'explorer cette évolution, même sous une forme optionnelle ou configurable ?

A noter que si nos échanges amènent à une solution intéressante à implémenter, et si mon activité me le permets, je peux contribuer au projet pour l'implémentation.

Merci pour votre retour.

**Reply by dbarzin** [reply]
Si je comprends bien, la demande est de pouvoir lier un processus à un service applicatif ?

**Reply by j-ova** [reply]
Le processus étant trop haut niveau, l'idée serait plutôt de pouvoir lier les sous-niveaux d'un processus à une application.

Ce qui semble cohérent quand on le formule à voix haute, c'est de se dire qu'une Tâche (ie. une "activité élémentaire" selon la définition de l'ANSSI) est réalisée à l'aide d'une ou plusieurs applications, quitte à déclarer 1 opération = 1 tâche si l'opération est simple.

**Comment by MCF241** [comment]
Bonjour, Désolé d'intervenir dans la discussion.

Pour moi le soucis que j'ai est sur le vocabulaire et l'ordre des choses. Dans notre cartographie "Officielle" nous avons recensé beaucoup de processus avec, il me semble, une grande confusion. Cela fait quelque temps que je travail sur cette problématique et il semblerais que le vocabulaire est bien le point bloquant. Je viens du BPMn V2 est je modélise depuis quelques années insouciant et joyeux...

Mais au moment de transferer de l'officielle vers dans Mercator je me suis retrouvé avec 250 Macro Processus (Il y en a 5 voir 7 max).
Des centaines de processus et pas d'activité, pas de tâches les acteurs inconnus au bataillon et quelque informations.

La confusion viens de cette organisation et du sens que l'on donne en fonction de notre historique (Tech, métier etc.).

<img width="1404" alt="image" src="https://github.com/user-attachments/assets/0ea13f39-fe58-4bfa-946f-ce137efac27a" />

Dans le schéma (qui est incorrect à mon avis) on vois qu'il y a des non sens, quid d'une activité, quid d'une opération, tout est mélangé (les joies des import export).

Donc le canvas est les suivant:

Il y a bien 5 Macro processus (Dans l'enseignement supérieur et recherche):

1. Processus de pilotage
2. Processus de support
3. Processus opérationnels
4. Processus de valorisation
5. Processus de Recherche (pour l'ESR)

D'aprés mes collègues il y a bien 250 processus identifiés à l'Université (ils glissent donc de Macro a Processus dans Mercator) il reste donc a qualifié qu'es-ce une Activité d'ou découlent des Opérations et des Tâches.

Si je compare donc au BPMn v2:

Les Macro Processus sont pour les pilotage, il n'y a pas d'équivalent en BPM (_Ormis le design des chemin parfaits ???_)

- Processus = Piscine (pool)
- Les acteurs restent les acteurs = Lignes de nages (swimlanes)
- Activités = Sous processus
- Opération = Tâche automatique (Selon Didier: _Opérations : Correspondent à des étapes spécifiques dans un processus._)
- Tâche = Activité (Humaine ou Automatique) 

Pour moi je n'ai pas encore bien réussi à dissocier Activité et Tâches ou Opération/Tâches.
